### PR TITLE
Add filters for subtotalSuffix and saleBadgeSuffix

### LINKED ISSUE #3747: Integration Tracking (WC Subscriptions): Add recurring period to subtotal price and saving
Cart and Checkout items need to have a recurring period appended to them, this is for both the sale badge and item subtotal, but not the item total.

image
image

### PR DESCRIPTION
**Pull request by opr**
### Based against `fix/3735-filters-api`
- In `CartLineItemRow` execute the `subtotalPriceFormat` filter which extensions can use to to format the `ProductPrice` element that will be displayed under the product name. It must return a string containing `<price/>` so it can be used in `createInterpolateElement`.
- Modify `ProductPrice` so that it outputs the price in a way that follows the `format` it was passed.
- In `CartLineItemRow` execute the `saleBadgePriceFormat` filter which extensions can use to modify the _price_ of the `SaleBadge` component. It does not allow for the `Save` word to be changed currently.
- Add `format` prop to the `SaleBadge` component.
- Render the price in `SaleBadge` using `createInterpolateElement`
- Add tests for `ProductPrice` to test passing a format.
- 

Fixes #3747
Replaces #3763

#### Before
image

#### After
image

#### Before 
image

#### After
image

### How to test the changes in this Pull Request:

1. Check out add/filter-subtotal-string-2 in blocks.
2. Add a non-subscription product to your cart with a sale price.
3. Add a subscription product to your cart with a sale price.
4. Add varying lengths/types of subscription product to your cart, every 2 weeks, every 6 months etc.
5. Open the cart block and verify the sale badge shows the period for subscriptions products, but not for regular products.
6. Verify the subtotal for subscription products shows under each product name and contains the period.
7. Verify the non-subscription product does not show a period for the subtotal.

### CODE REVIEW

File: assets/js/base/components/cart-checkout/order-summary/order-summary-item.js [hunk 1]
```diff
@@ -50,6 +51,15 @@ const OrderSummaryItem = ( { cartItem } ) => {
...
 		.convertPrecision( currency.minorUnit )
 		.getAmount();
+	const subtotalPriceFormat = __experimentalApplyCheckoutFilter( {
+		filterName: 'subtotalPriceFormat',
+		defaultValue: '',
+		arg: {
+			lineItem: cartItem,
+		},
+		// Only accept strings.
+		validation: ( value ) => typeof value === 'string',
```
Comment (Aljullu) [inline assets/js/base/components/cart-checkout/order-summary/order-summary-item.js]: Should we check that it includes `<price/>` too?

**Review by budzanowski — APPROVED** [review]
Looks good.

**Review by Aljullu — APPROVED** [review]
LGTM as well. There is a small visual glitch with the Sale badge under the subtotal overflowing the table:

imatge

File: assets/js/base/components/cart-checkout/order-summary/order-summary-item.js (hunk 1 above)
Comment (opr) [reply assets/js/base/components/cart-checkout/order-summary/order-summary-item.js]: Good catch, thanks Albert

**Comment by opr** [comment]
> LGTM as well. There is a small visual glitch with the Sale badge under the subtotal overflowing the table:
> 
> imatge

Thanks, I had fixed this in the last PR but forgot to copy the CSS changes over into this one. Updated now.
image

**Review by Aljullu — APPROVED** [review]
Looks good now, thanks @opr8!

**Pull request merged**